components/src/preact/mutationFilter/mutation-filter.tsx

```diff
@@ -262,82 +261,69 @@ function getPlaceholder(referenceGenome: ReferenceGenome) {
+            return singleGraphColorRGBByName('purple', alpha);
+    }
 };
 
 type SelectedFilterProps = {
-    handleRemoveValue: (mutation: ParsedMutationFilter) => void;
-    mutationFilter: ParsedMutationFilter;
+    handleRemoveValue: (mutation: MutationFilterItem) => void;
+    mutationFilter: MutationFilterItem;
 };
 
 const SelectedFilter = ({ handleRemoveValue, mutationFilter }: SelectedFilterProps) => {
     return (
         <span
             key={mutationFilter.value.toString()}
-            className='center p-2 m-1 inline-flex text-black rounded-md'
+            className='center px-2 py-1 inline-flex text-black rounded-md'
             style={{
                 backgroundColor: backgroundColorMap(mutationFilter),
             }}
         >
             {mutationFilter.value.toString()}
-            <button className='ml-1 cursor-pointer' onClick={() => handleRemoveValue(mutationFilter)}>
+            <button
+                className='ml-1 cursor-pointer'
+                aria-label={`remove mutation filter ${mutationFilter.value.code}`}
+                onClick={() => handleRemoveValue(mutationFilter)}
+            >
                 ×
             </button>
         </span>
     );
 };
 
-function mapToMutationFilterStrings(selectedFilters: SelectedFilters) {
-    return {
-        aminoAcidMutations: selectedFilters.aminoAcidMutations.map((mutation) => mutation.toString()),
-        nucleotideMutations: selectedFilters.nucleotideMutations.map((mutation) => mutation.toString()),
-        aminoAcidInsertions: selectedFilters.aminoAcidInsertions.map((insertion) => insertion.toString()),
-        nucleotideInsertions: selectedFilters.nucleotideInsertions.map((insertion) => insertion.toString()),
-    };
+function mapToMutationFilterStrings(selectedFilters: MutationFilterItem[]) {
+    return selectedFilters.reduce(
+        (acc, filter) => {
+            switch (filter.type) {
+                case 'nucleotideMutations':
+                    return { ...acc, nucleotideMutations: [...acc.nucleotideMutations, filter.value.toString()] };
+                case 'aminoAcidMutations':
+                    return { ...acc, aminoAcidMutations: [...acc.aminoAcidMutations, filter.value.toString()] };
+                case 'nucleotideInsertions':
+                    return { ...acc, nucleotideInsertions: [...acc.nucleotideInsertions, filter.value.toString()] };
+                case 'aminoAcidInsertions':
+                    return { ...acc, aminoAcidInsertions: [...acc.aminoAcidInsertions, filter.value.toString()] };
+            }
+        },
+        {
+            aminoAcidMutations: [],
+            nucleotideMutations: [],
+            aminoAcidInsertions: [],
+            nucleotideInsertions: [],
+        } as {
+            aminoAcidMutations: string[];
+            nucleotideMutations: string[];
+            aminoAcidInsertions: string[];
+            nucleotideInsertions: string[];
+        },
```

Comment:
```suggestion
        {
            aminoAcidMutations: [],
            nucleotideMutations: [],
            aminoAcidInsertions: [],
            nucleotideInsertions: [],
        } as MutationsFilter,
```

We could even get rid of the `as` by adding the type either as return type of the method or as generic argument to `reduce`.